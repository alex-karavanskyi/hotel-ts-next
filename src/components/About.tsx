'use client'
import React from 'react'
import { motion } from 'framer-motion'
import { styles } from '../styles'
import { services } from '../constants'
import { fadeIn, textVariant } from '../utils/motion'
import { ServiceCard } from '.'

const About = React.forwardRef<HTMLDivElement>((props, ref) => {
  return (
    <div
      id='about'
      className={`${styles.padding} max-w-7xl mx-auto relative z-0 my-7`}
    >
      <motion.div variants={textVariant(0.1)}>
        <p className={styles.sectionSubText}>Introduction</p>
        <h2 className={styles.sectionHeadText}>Overview.</h2>
      </motion.div>

      <motion.p
        variants={fadeIn('', '', 0.1, 1)}
        className='mt-4 text-[17px] max-w-3xl leading-[30px] text-white-100'
      >
        I'm a skilled software developer with experience in TypeScript and
        JavaScript, and expertise in frameworks like React, Node.js, and
        Three.js. I'm a quick learner and collaborate closely with clients to
        create efficient, scalable, and user-friendly solutions that solve
        real-world problems. Let's work together to bring your ideas to life!
      </motion.p>
      <div className='mt-20 grid grid-cols-2 gap-10'>
        {services.map((service, index) => (
          <ServiceCard key={service.title} index={index} {...service} />
        ))}
      </div>
    </div>
  )
})

export default About
