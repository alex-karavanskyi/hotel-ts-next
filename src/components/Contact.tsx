'use client'
import { useForm } from 'react-hook-form'
import { motion } from 'framer-motion'
import { slideIn } from '../utils/motion'

type FormData = {
  name: string
  email: string
  message: string
}

const Contact = () => {
  const {
    register,
    formState: { errors, isValid },
    handleSubmit,
    reset,
  } = useForm<FormData>({ mode: 'onChange' })

  const onSubmit = (data: FormData) => {
    alert(JSON.stringify(data))
    reset()
  }

  return (
    <div className='contact-container' id='contact'>
      <motion.div
        variants={slideIn('left', 'tween', 0.2, 1)}
        className='contact-flex'
      >
        <p>Get in touch</p>
        <h3 style={{ color: 'white' }}>Contact.</h3>

        <form onSubmit={handleSubmit(onSubmit)} className='contact-form'>
          <label>
            <span>Name</span>
            <input
              {...register('name', {
                required: true,
                minLength: {
                  value: 3,
                  message: 'Name must be min 3 characters',
                },
              })}
            />
          </label>
          <div>
            {errors.name && (
              <p style={{ color: 'red', marginBottom: 0 }}>
                {errors.name?.message}
              </p>
            )}
          </div>
          <label>
            <span>Email</span>
            <input
              {...register('email', {
                required: true,
                minLength: {
                  value: 3,
                  message: 'Email must be min 3 characters',
                },
              })}
            />
          </label>
          <div>
            {errors.email && (
              <p style={{ color: 'red', marginBottom: 0 }}>
                {errors.email?.message}
              </p>
            )}
          </div>
          <label>
            <span>Message</span>
            <textarea
              rows={7}
              {...register('message', {
                required: true,
                minLength: {
                  value: 15,
                  message: 'Message must be min 15 characters',
                },
              })}
            />
          </label>
          <div>
            {errors.message && (
              <p style={{ color: 'red', marginBottom: 0 }}>
                {errors.message?.message}
              </p>
            )}
          </div>
          <button
            type='submit'
            className={`contact-button ${
              isValid ? 'contact-button-black' : 'contact-button-white'
            }`}
            disabled={!isValid}
          >
            Send
          </button>
        </form>
      </motion.div>
    </div>
  )
}

export default Contact
