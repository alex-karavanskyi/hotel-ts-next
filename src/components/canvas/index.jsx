'use client'
import StarsCanvas from './Stars'
import LaptopCanvas from './Laptop'

export { StarsCanvas, LaptopCanvas }
