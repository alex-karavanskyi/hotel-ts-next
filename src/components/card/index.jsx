import ExperienceCard from './ExperienceCard'
import ProjectCard from './ProjectCard'
import ServiceCard from './ServiceCard'

export { ExperienceCard, ProjectCard, ServiceCard }
