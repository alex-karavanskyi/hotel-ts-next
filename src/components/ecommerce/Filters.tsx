'use client'
import styled from 'styled-components'
import { useEffect, useState, memo } from 'react'
import { useAppSelector, useAppDispatch } from '@/redux/hooks'
import { getUniqueValues, formatPrice } from '@/utils/helpers'
import { numberPagination } from '@/redux/features/paginationSlice'
import {
  clearFilters,
  updateFilters,
  loadProducts,
  filterProducts,
} from '@/redux/features/filterSlice'

type HandleValue = string | number

const Filters = () => {
  const {
    filters: { text, category, min_price, price, max_price },
    all_products,
  } = useAppSelector((store) => store.filter)
  const [buttonColor, setButtonColor] = useState<string>(category)
  const { products } = useAppSelector((store) => store.products)

  const dispatch = useAppDispatch()

  const categories = getUniqueValues(all_products, 'category')

  const handleFilters = (
    e: React.FormEvent<HTMLInputElement> | React.MouseEvent<HTMLButtonElement>
  ) => {
    let name = e.currentTarget.name
    let value: HandleValue = e.currentTarget.value
    if (name === 'category') {
      value = e.currentTarget.textContent || ''
      setButtonColor(value)
    }
    if (name === 'price') {
      value = Number(value)
    }
    dispatch(updateFilters({ name, value }))
    dispatch(numberPagination(1))
  }

  useEffect(() => {
    dispatch(loadProducts(products))
  }, [products, dispatch])

  useEffect(() => {
    dispatch(filterProducts())
  }, [category, price, text, dispatch])

  return (
    <Wrapper>
      <form onSubmit={(e) => e.preventDefault()}>
        <div className='form-control-container'>
          <input
            type='text'
            name='text'
            placeholder='search'
            className='search-input'
            value={text}
            onChange={handleFilters}
          />
          {categories.map((c, index) => {
            if (typeof c === 'string') {
              return (
                <button
                  key={index}
                  onClick={handleFilters}
                  type='button'
                  name='category'
                  style={{
                    border:
                      buttonColor === c
                        ? 'solid 2px black'
                        : 'var(--clr-grey-5)',
                  }}
                >
                  {c}
                </button>
              )
            }
            return null
          })}
        </div>
        <div className='form-control-price'>
          <h5>price</h5>
          <p className='price'>{formatPrice(price)}</p>
          <input
            type='range'
            name='price'
            min={min_price}
            max={max_price}
            value={price}
            onChange={handleFilters}
          />
        </div>
      </form>
      <button
        type='button'
        className='clear-btn'
        onClick={() => dispatch(clearFilters())}
      >
        clear filters
      </button>
    </Wrapper>
  )
}

const Wrapper = styled.section`
  margin: 0 auto;
  margin-bottom: 50px;
  max-width: 1248px;
  .search-input {
    padding-left: 0.5rem;
    margin-right: 1rem;
    background: var(--clr-grey-10);
    border-radius: var(--radius);
    border-color: transparent;
    letter-spacing: var(--spacing);
  }
  .search-input::placeholder {
    text-transform: capitalize;
  }
  .form-control-container button {
    display: grid;
    grid-template-columns: 118px;
    text-transform: capitalize;
    background: transparent;
    border: none;
    letter-spacing: var(--spacing);
    color: var(--clr-grey-5);
    cursor: pointer;
    padding: 14px;
  }
  .form-control-container {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
  }
  .form-control-price {
    padding-top: 30px;
    margin-left: 14px;
  }
  .clear-btn {
    padding-top: 14px;
    margin-left: 14px;
    text-transform: capitalize;
    background: transparent;
    border: none;
    letter-spacing: var(--spacing);
    color: var(--clr-grey-5);
    cursor: pointer;
  }
`

export default memo(Filters)
