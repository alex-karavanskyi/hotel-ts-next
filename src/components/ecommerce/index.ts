import Sort from './filters/Sort'
import Search from './filters/Search'
import Price from './filters/Price'
import Category from './filters/Category'
import ProductControls from './filters/ProductControls'
import GridView from './list/GridView'
import ListView from './list/ListView'
import Pagination from './list/Pagination'
import ProductList from './list/ProductList'
import ProductImages from './single/ProductImages'

export {
  ProductList,
  ProductControls,
  ProductImages,
  GridView,
  ListView,
  Pagination,
  Sort,
  Search,
  Price,
  Category,
}
