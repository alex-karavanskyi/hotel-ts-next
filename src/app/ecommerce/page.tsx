import { Filters, ProductList, Sort } from '@/components/ecommerce'
import { Metadata } from 'next'

export const metadata: Metadata = {
  title: 'Rooms | React App',
}

const Rooms = () => {
  return (
    <div className='room-page'>
      <div className='room-center'>
        <h1>E-COMMERCE</h1>
        <div>
          REFINED, SOPHISTICATED, SMART AND SIMPLE, YET FLAMBOYANT AND
          PROVOCATIVE, ALL AT THE SAME TIME. UNIVERSALLY APPEALING HOTEL ROOMS
          ARE THOUGHTFULLY DESIGNED, DOWN TO EVERY LAST FINISH AND DETAIL, WITH
          DISTINCT ZONES FOR SLEEPING, RELAXING, WORKING AND ENTERTAINING. WE'VE
          INTEGRATED SEAMLESS SMART HOTEL TECHNOLOGY SO EVERYTHING YOU NEED IS
          WITHIN ARM'S REACH AND RIGHT AT YOUR FINGERTIPS. A STAY AT NYC'S
          PUBLIC HOTEL OFFERS ALL THE COMFORTS OF HOME, BUT BETTER.
        </div>
      </div>
      <Filters />
      <Sort />
      <ProductList />
    </div>
  )
}

export default Rooms
