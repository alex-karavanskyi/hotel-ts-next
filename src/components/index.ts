'use client'

import Footer from './Footer'
import Navbar from './Navbar'
import Sidebar from './Sidebar'
import Filters from './Filters'
import Sort from './Sort'
import ListView from './ListView'
import GridView from './GridView'
import ProductList from './ProductList'
import Error from './Error'
import Loading from './Loading'
import ProductImages from './ProductImages'
import ProductSlider from './ProductSlider'
import ServiceCard from './card/ServiceCard'
import ExperienceCard from './card/ExperienceCard'
import Hero from './Hero'

export {
  Navbar,
  Footer,
  Sidebar,
  Filters,
  Sort,
  ProductList,
  ListView,
  GridView,
  Error,
  Loading,
  ProductImages,
  ProductSlider,
  Hero,
  ServiceCard,
  ExperienceCard,
}
