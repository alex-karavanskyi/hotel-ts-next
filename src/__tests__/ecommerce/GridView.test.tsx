import { screen, render } from '@testing-library/react'
import { GridView } from '@/components/ecommerce'
import { Provider } from 'react-redux'
import { configureStore } from '@reduxjs/toolkit'
import productReducer from '@/redux/features/productSlice'
import favoriteReducer from '@/redux/features/favoriteSlice'

describe('GridView', () => {
  const createTestStore = () =>
    configureStore({
      reducer: {
        favorite: favoriteReducer,
        products: productReducer,
      },
    })
  it('render correctly', () => {
    const store = createTestStore()

    render(
      <Provider store={store}>
        <GridView
          products={[
            {
              id: 'rec69hj6QLF5zEjue',
              name: 'Rowenta X-Plorer',
              price: 477,
              description:
                'Cloud bread VHS hell of banjo bicycle rights jianbing umami mumblecore etsy 8-bit pok pok +1 wolf. Vexillologist yr dreamcatcher waistcoat, authentic chillwave trust fund. Viral typewriter fingerstache pinterest pork belly narwhal. Schlitz venmo everyday carry kitsch pitchfork chillwave iPhone taiyaki trust fund hashtag kinfolk microdosing gochujang live-edge',
              category: 'Misc',
              image:
                'https://v5.airtableusercontent.com/v3/u/36/36/1735927200000/z1DGYY2OwoRZbvhrrDDN7A/w1Mz_ziHyacFNEWhG20n9-9402zZeOrO_8Hl08ziofX1k04NPQxGz5Bpukazy2Y-vDBOH0WckJtDMDsT97U5ULKsHXAbW3O0f7E7DK28kuUh325yvhA1LacIvbnNSrhcR6gktaXkjthLToYH2twKRNX4yLJvD7rSvq1DK9pafxU/9_ssyMBUu_Cn3gwqDuN_3fojw58rg4Lyfag_pangujw',
              images: [],
            },
            {
              id: 'rec6b9Eg8wLI02z6s',
              name: 'Roborock S8',
              price: 575,
              description:
                'Cloud bread VHS hell of banjo bicycle rights jianbing umami mumblecore etsy 8-bit pok pok +1 wolf. Vexillologist yr dreamcatcher waistcoat, authentic chillwave trust fund. Viral typewriter fingerstache pinterest pork belly narwhal. Schlitz venmo everyday carry kitsch pitchfork chillwave iPhone taiyaki trust fund hashtag kinfolk microdosing gochujang live-edge',
              category: 'Misc',
              image:
                'https://v5.airtableusercontent.com/v3/u/36/36/1735927200000/YieCsTEpUT8lDPRjqfcYWg/CHkgL9-0l9AvGrfo94YdgqNGJaghkmlgHd_YXwG1QAP2iSkTfqWSl6MsIebct5DtdTHaZ6xty5HI3WU3SjfI_EcoTmS2_s9tbRBTLIhjbPS_VcgB3bG3mECbnkz8cKqaPefFwOjUGgnNWN5g62uM4o_d1J-r7hHxEbNpBwvzLBY/psMEKs-Azx8sqC_ldD59loQ0GCp9su67li579XWJdgE',
              images: [],
            },
          ]}
        />
      </Provider>
    )
    expect(screen.getByRole('list')).toMatchSnapshot()
  })
})
