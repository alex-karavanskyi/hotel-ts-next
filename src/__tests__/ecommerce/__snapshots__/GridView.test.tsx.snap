// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridView render correctly 1`] = `
<div
  class="grid__view-products"
  role="list"
>
  <article>
    <div
      class="grid__view-products-images"
    >
      <img
        alt="Rowenta X-Plorer"
        class="grid__view-images"
        data-nimg="1"
        decoding="async"
        height="500"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fv5.airtableusercontent.com%2Fv3%2Fu%2F36%2F36%2F1735927200000%2Fz1DGYY2OwoRZbvhrrDDN7A%2Fw1Mz_ziHyacFNEWhG20n9-9402zZeOrO_8Hl08ziofX1k04NPQxGz5Bpukazy2Y-vDBOH0WckJtDMDsT97U5ULKsHXAbW3O0f7E7DK28kuUh325yvhA1LacIvbnNSrhcR6gktaXkjthLToYH2twKRNX4yLJvD7rSvq1DK9pafxU%2F9_ssyMBUu_Cn3gwqDuN_3fojw58rg4Lyfag_pangujw&w=1080&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fv5.airtableusercontent.com%2Fv3%2Fu%2F36%2F36%2F1735927200000%2Fz1DGYY2OwoRZbvhrrDDN7A%2Fw1Mz_ziHyacFNEWhG20n9-9402zZeOrO_8Hl08ziofX1k04NPQxGz5Bpukazy2Y-vDBOH0WckJtDMDsT97U5ULKsHXAbW3O0f7E7DK28kuUh325yvhA1LacIvbnNSrhcR6gktaXkjthLToYH2twKRNX4yLJvD7rSvq1DK9pafxU%2F9_ssyMBUu_Cn3gwqDuN_3fojw58rg4Lyfag_pangujw&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fv5.airtableusercontent.com%2Fv3%2Fu%2F36%2F36%2F1735927200000%2Fz1DGYY2OwoRZbvhrrDDN7A%2Fw1Mz_ziHyacFNEWhG20n9-9402zZeOrO_8Hl08ziofX1k04NPQxGz5Bpukazy2Y-vDBOH0WckJtDMDsT97U5ULKsHXAbW3O0f7E7DK28kuUh325yvhA1LacIvbnNSrhcR6gktaXkjthLToYH2twKRNX4yLJvD7rSvq1DK9pafxU%2F9_ssyMBUu_Cn3gwqDuN_3fojw58rg4Lyfag_pangujw&w=1080&q=75 2x"
        style="color: transparent;"
        width="470"
      />
      <a
        class="grid__view-products-link"
        href="/ecommerce/rec69hj6QLF5zEjue"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          />
        </svg>
      </a>
    </div>
    <footer
      class="grid__view-footer"
    >
      <h5
        class="grid__view-favorite"
      >
        <div>
          Rowenta X-Plorer
           
          <svg
            class="grid__view-favorite-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
            />
          </svg>
        </div>
      </h5>
      <p
        class="grid__view-products-price"
      >
        $477.00
      </p>
    </footer>
  </article>
  <article>
    <div
      class="grid__view-products-images"
    >
      <img
        alt="Roborock S8"
        class="grid__view-images"
        data-nimg="1"
        decoding="async"
        height="500"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fv5.airtableusercontent.com%2Fv3%2Fu%2F36%2F36%2F1735927200000%2FYieCsTEpUT8lDPRjqfcYWg%2FCHkgL9-0l9AvGrfo94YdgqNGJaghkmlgHd_YXwG1QAP2iSkTfqWSl6MsIebct5DtdTHaZ6xty5HI3WU3SjfI_EcoTmS2_s9tbRBTLIhjbPS_VcgB3bG3mECbnkz8cKqaPefFwOjUGgnNWN5g62uM4o_d1J-r7hHxEbNpBwvzLBY%2FpsMEKs-Azx8sqC_ldD59loQ0GCp9su67li579XWJdgE&w=1080&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fv5.airtableusercontent.com%2Fv3%2Fu%2F36%2F36%2F1735927200000%2FYieCsTEpUT8lDPRjqfcYWg%2FCHkgL9-0l9AvGrfo94YdgqNGJaghkmlgHd_YXwG1QAP2iSkTfqWSl6MsIebct5DtdTHaZ6xty5HI3WU3SjfI_EcoTmS2_s9tbRBTLIhjbPS_VcgB3bG3mECbnkz8cKqaPefFwOjUGgnNWN5g62uM4o_d1J-r7hHxEbNpBwvzLBY%2FpsMEKs-Azx8sqC_ldD59loQ0GCp9su67li579XWJdgE&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fv5.airtableusercontent.com%2Fv3%2Fu%2F36%2F36%2F1735927200000%2FYieCsTEpUT8lDPRjqfcYWg%2FCHkgL9-0l9AvGrfo94YdgqNGJaghkmlgHd_YXwG1QAP2iSkTfqWSl6MsIebct5DtdTHaZ6xty5HI3WU3SjfI_EcoTmS2_s9tbRBTLIhjbPS_VcgB3bG3mECbnkz8cKqaPefFwOjUGgnNWN5g62uM4o_d1J-r7hHxEbNpBwvzLBY%2FpsMEKs-Azx8sqC_ldD59loQ0GCp9su67li579XWJdgE&w=1080&q=75 2x"
        style="color: transparent;"
        width="470"
      />
      <a
        class="grid__view-products-link"
        href="/ecommerce/rec6b9Eg8wLI02z6s"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          />
        </svg>
      </a>
    </div>
    <footer
      class="grid__view-footer"
    >
      <h5
        class="grid__view-favorite"
      >
        <div>
          Roborock S8
           
          <svg
            class="grid__view-favorite-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
            />
          </svg>
        </div>
      </h5>
      <p
        class="grid__view-products-price"
      >
        $575.00
      </p>
    </footer>
  </article>
</div>
`;
